src/scipp/coords/transform_coords.py

```diff
@@ -150,29 +150,36 @@ def _log_transform(rules: List[Rule], targets: Set[str],
 
 
 def _store_coord(da: DataArray, name: str, coord: Coord) -> None:
-    def out(x):
-        return x.coords if coord.destination == Destination.coord else x.attrs
-
-    def del_other(x):
-        try:
-            if coord.destination == Destination.coord:
-                del x.attrs[name]
-            else:
-                del x.coords[name]
-        except NotFoundError:
-            pass
-
-    if coord.has_dense:
-        out(da)[name] = coord.dense
-        del_other(da)
-    if coord.has_event:
-        try:
-            out(da.bins)[name] = coord.event
-        except VariableError:
-            # Thrown on mismatching bin indices, e.g. slice
-            da.data = da.data.copy()
-            out(da.bins)[name] = coord.event
-        del_other(da.bins)
+    def try_del(dest):
+        if dest == Destination.coord:
+            da.coords.pop(name, None)
+            if da.bins is not None:
+                da.bins.coords.pop(name, None)
+        else:
+            da.attrs.pop(name, None)
+            if da.bins is not None:
+                da.bins.attrs.pop(name, None)
```

Comment:
I don't think there is a guarantee that either both of the bin and event items are coords or attrs? One may be coord, the other an attr?

Reply:
That would be strange. But it would work. Let's say, `coord.destination=attr` and the events had a coord but the bins an attr. The first call to `try_del` removes the event coord but leaves the bin attr. Then, if `usages=0`, the bin attr is removed. Else, the bin attr is overwritten by `coord.dense` and `coord.event` is stored as an attr of events. So everything is fine.
I think all other cases work correspondingly. 

Is this a relevant case? Should I add unit tests?

Reply:
> Is this a relevant case? Should I add unit tests?

Actually I may happen quite easily: Consider `da` with bin as well as event coord `x`. Then
```python
da['x', 0]
```
will have an `x` attr but `da.bins` will have an `x` coord.